### PR TITLE
Merge of 2.8 features to master

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

**Review by alonre24 — COMMENTED** [review]
Few small comments

File: tests/pytests/test_async.py [hunk 1]
```diff
@@ -74,3 +70,38 @@ def test_mod4745(env):
     # Make sure we are getting here without having cluster mark itself as fail since the server is not responsive and
     # fail to send cluster PING on time before we reach cluster-node-timeout.
     waitForIndex(r, 'idx')
+
+def test_eval_node_errors_async(env):
+    if not POWER_TO_THE_WORKERS:
```
[the remark this answers is not in the document]
Comment (rafie) [reply tests/pytests/test_async.py:75]: This follows: `env = Env(moduleArgs='DEFAULT_DIALECT 2 WORKER_THREADS 1 ENABLE_THREADS TRUE ON_TIMEOUT FAIL')`
So, if not `power` then skip.

File: coord/src/config.h [hunk 2]
```diff
@@ -29,7 +30,11 @@ extern SearchClusterConfig clusterConfig;
 
 #define DEFAULT_CLUSTER_CONFIG                                                             \
   (SearchClusterConfig) {                                                                  \
-    .numPartitions = 0, .type = DetectClusterType(), .timeoutMS = 500, .globalPass = NULL, \
+    .numPartitions = 0,                                                                    \
+    .connPerShard = 0,                                                                     \
```
[the remark this answers is not in the document]
Comment (GuyAv46) [reply coord/src/config.h:34]: on coordinator's init

File: src/aggregate/aggregate_exec.c [hunk 3]
```diff
@@ -439,16 +549,37 @@ static int execCommandCommon(RedisModuleCtx *ctx, RedisModuleString **argv, int
...
+    workersThreadPool_AddWork((redisearch_thpool_proc)AREQ_Execute_Callback, BCRctx);
   } else {
-    if (IsProfile(r)) {
-      RedisModule_ReplyWithArray(ctx, 2);
+    if (prepareExecutionPlan(r, AREQ_BUILDPIPELINE_NO_FLAGS, &status) != REDISMODULE_OK) {
+      goto error;
     }
     AREQ_Execute(r, ctx);
   }
+#else
```
Comment (GuyAv46) [inline src/aggregate/aggregate_exec.c:575]: why do we need the `else` here? its the same code as above

File: src/aggregate/aggregate_request.c [hunk 4]
```diff
@@ -1,1433 +1,1565 @@
...
+  // check possible optimization after creation of QueryNode tree
+  OPTMZ(QOptimizer_QueryNodes(req->ast.root, req->optimizer));
+
+  ConcurrentSearchCtx_Init(sctx->redisCtx, &req->conc);
+
+  // moved
+  //req->rootiter = QAST_Iterate(ast, opts, sctx, &req->conc, req->reqflags, status);
+
+  // check possible optimization after creation of IndexIterator tree
+  //OPTMZ(QOptimizer_Iterators(req, req->optimizer));
```
Comment (GuyAv46) [inline src/aggregate/aggregate_request.c:914]: should be deleted

File: src/inverted_index.c [hunk 5]
```diff
@@ -848,6 +848,15 @@ SKIPPER(seekRawDocIdsOnly) {
   // we cannot get out of range since we check in
   if (curVal < delta) {
     cur++;
+
+#if 1
```
Comment (GuyAv46) [inline src/inverted_index.c:852]: why if 1?

File: src/redisearch.h [hunk 6]
```diff
@@ -302,6 +303,7 @@ typedef struct RSIndexResult {
    *******************************************************************************/
   /* The docId of the result */
   t_docId docId;
+  const RSDocumentMetadata *dmd;
```
Comment (GuyAv46) [inline src/redisearch.h:306]: this addition here violates the `IMPORTANT` comment above. Should we move it to another part of the struct?

**Review by alonre24 — DISMISSED** [review]

**Pull request merged**